woocommerce/woocommerce
# Multichannel Marketing - Core Library

### All Submissions:






### Changes proposed in this Pull Request:



This PR implements the core libraries for the multichannel marketing feature (#34548). Third-party extensions can utilize the code introduced in this PR to register themselves as a marketing channel. The registered marketing channels will be displayed on the "Marketing" dashboard under "Installed Extensions" once activated.

It adds a new interface called `MarketingChannelInterface`, which third-party marketing extensions can implement to give the core access to their internal data (e.g., status, ads campaigns, etc.). The third-party extensions can call the `MarketingChannels::register` method or the `woocommerce_marketing_channels` filter to register themselves.

Currently, we limit the list of marketing channels to the one returned by the "Recommended Plugins" API endpoint in WooCommerce.com (`GET https://woocommerce.com/wp-json/wccom/marketing-tab/1.2/recommendations.json`). Only the extensions returned by that API can register as a marketing channel. If the channel's slug does not match any of the extension slugs returned by that API, we will silently log an error and skip that channel.

Previously, the user could see installed but deactivated extensions in their "Installed Extensions" list; This was possible because we used a hardcoded list to compile the extensions in that list. However, with this change applied, only the activated extensions will appear on the "Installed Extensions" list. This also means that the user won't be able to "Activate" any extensions through the marketing page.

Closes #34549.

### How to test the changes in this Pull Request:

**Note:** To test this PR, you need to install and active the [Google Listings & Ads](https://github.com/woocommerce/google-listings-and-ads) extension using its repo branch [feature/mcm-integration](https://github.com/woocommerce/google-listings-and-ads/tree/feature/mcm-integration) (related PR: https://github.com/woocommerce/google-listings-and-ads/pull/1730).

1. Go to `WooCommerce Settings > Advanced > Features` and turn on the multichannel marketing feature.
2. Go to the Marketing > Overview page and confirm Google Listings & Ads appears in the "Installed marketing extensions" section
3. Ensure you have not set up the extension or completed onboarding. If you have, reset the extension data.
4. Confirm that a link to "Finish setup" is displayed next to each extension name if it has not been set up and links to their onboarding/setup process.
5. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with the "Settings" link and points to the extension's settings page.


### Other information:

-   [ ] Have you created a changelog file for each project being changed, i.e.` pnpm --filter=<project> run changelog add`?



### FOR PR REVIEWER ONLY:
